certinfo.go

```diff
@@ -749,20 +844,218 @@ func CertificateRequestText(csr *x509.CertificateRequest) (string, error) {
+					buf.WriteString(fmt.Sprintf("%16sCA:FALSE", ""))
+				}
+				if constraints.MaxPathLen == 0 {
+					buf.WriteString(fmt.Sprintf(", pathlen:0\n"))
+				} else if constraints.MaxPathLen > 0 {
+					buf.WriteString(fmt.Sprintf(", pathlen:%d\n", constraints.MaxPathLen))
+				} else {
+					buf.WriteString("\n")
+				}
+			case ext.Id.Equal(oidExtNameConstraints):
+				// RFC 5280, 4.2.1.10
+				// NameConstraints ::= SEQUENCE {
+				//      permittedSubtrees       [0]     GeneralSubtrees OPTIONAL,
+				//      excludedSubtrees        [1]     GeneralSubtrees OPTIONAL }
+				//
+				// GeneralSubtrees ::= SEQUENCE SIZE (1..MAX) OF GeneralSubtree
+				//
+				// GeneralSubtree ::= SEQUENCE {
+				//      base                    GeneralName,
+				//      minimum         [0]     BaseDistance DEFAULT 0,
+				//      maximum         [1]     BaseDistance OPTIONAL }
+				//
+				// BaseDistance ::= INTEGER (0..MAX)
+				var constraints nameConstraints
+				_, err := asn1.Unmarshal(ext.Value, &constraints)
+				if err != nil {
+					continue
+				}
+				if len(constraints.Excluded) > 0 && ext.Critical {
+					continue
+				}
+				var permittedDNSDomains []string
+				for _, subtree := range constraints.Permitted {
+					if len(subtree.Name) == 0 {
+						if ext.Critical {
+							continue
+						}
+						continue
+					}
+					permittedDNSDomains = append(permittedDNSDomains, subtree.Name)
+				}
+				buf.WriteString(fmt.Sprintf("%12sX509v3 Name Constraints:", ""))
+				if ext.Critical {
+					buf.WriteString(" critical\n")
+				} else {
+					buf.WriteString("\n")
+				}
+				if len(permittedDNSDomains) > 0 {
+					buf.WriteString(fmt.Sprintf("%16sPermitted:\n%18s%s", "", "", permittedDNSDomains[0]))
+					for i := 1; i < len(permittedDNSDomains); i++ {
+						buf.WriteString(fmt.Sprintf(", %s", permittedDNSDomains[i]))
+					}
+					buf.WriteString("\n")
+				}
+			case ext.Id.Equal(oidExtExtendedKeyUsage):
+				// extKeyUsage: RFC 5280, 4.2.1.12
+				// id-ce-extKeyUsage OBJECT IDENTIFIER ::= { id-ce 37 }
+				//
+				// ExtKeyUsageSyntax ::= SEQUENCE SIZE (1..MAX) OF KeyPurposeId
+				//
+				// KeyPurposeId ::= OBJECT IDENTIFIER
+				var keyUsage []asn1.ObjectIdentifier
+				_, err = asn1.Unmarshal(ext.Value, &keyUsage)
+				if err != nil {
+					continue
+				}
+				var eku []x509.ExtKeyUsage
+				for _, u := range keyUsage {
+					if extKeyUsage, ok := extKeyUsageFromOID(u); ok {
+						eku = append(eku, extKeyUsage)
+					}
+				}
+				buf.WriteString(fmt.Sprintf("%12sX509v3 Extended Key Usage:", ""))
+				if ext.Critical {
+					buf.WriteString(" critical\n")
+				} else {
+					buf.WriteString("\n")
+				}
+				var list []string
+				for _, val := range eku {
+					switch val {
+					case x509.ExtKeyUsageAny:
+						list = append(list, "Any Usage")
+					case x509.ExtKeyUsageServerAuth:
+						list = append(list, "TLS Web Server Authentication")
+					case x509.ExtKeyUsageClientAuth:
+						list = append(list, "TLS Web Client Authentication")
```

Comment:
I would drop `TLS Web` here.

Reply:
We use `TLS Web` for in the output for `step certificate inspect`. We should be consistent. Do you think it should be removed for both?

Reply:
Yes